### DISCUSSION TITLE
Consistent Build Failure: libtool issue

### DISCUSSION THREAD

**Discussion by WarthogARJ** [rbenv/ruby-build#2063] [Build failures]
Problem:
```
rbenv install 2.2.0
Installed openssl-1.0.2u to /Users/mbp_arj/.rbenv/versions/2.2.0
Downloading ruby-2.2.0.tar.bz2...
ruby-build: using readline from homebrew
ruby-build: using gmp from homebrew
BUILD FAILED (Mac OS X 10.15.7 using ruby-build 20220930-2-g55fb8a2)
```
My analysis: this is not a rbenv issue, but ruby-build problem
Others in the posted references have said the sanme thing.

This is perhaps a feature request, or at least a request for a properly document comment/workaround in the Installation istructions.
After I was unable to run "install rebenv 3.1.2 I tried all the trouble-shooting suggesting on rebenv.
As well as on ruby-build.
I then saw the EXACT same iproblem in various reported Issues, but they were never really Closed or acted upon.
But usually moved to "Discussion".

When I dug out the background, I followed what worked for them, and solved my problem.

Can the Mainteners of Ruby-Build, please discuss the issue with those of Rebenv, and make a Note that others can follow?
Ideally solve the issue, but having a documented workaround would help.

I was about to delete Rebenv and start using another package installer that would work until I dug up these more cryptic work-arounds.

### Steps to reproduce the behavior
Attempts to execute rebenv install 3.1.2 by numerous people including myself lead to failure

Refer to:
https://github.com/rbenv/ruby-build/issues/1505#issuecomment-752274233
https://bugs.ruby-lang.org/issues/17486
https://github.com/ruby/ruby/commit/ed01cc8fdc845b1db258eb3e35b3ba834463672e

I had EXACTLY these issues.
I solved it by following their advice:
(1) Writing the described patch
(2) Installing it in /var/tmp/configure.ac.patch
(3) brew upgrade
(4) brew install rbenv ruby-build libffi
(5) /var/tmp/configure.ac.patch rbenv install -p 3.1.2

Result: success

Install details:
```
Installing openssl-3.0.5...
Installed openssl-3.0.5 to /Users/mbp_arj/.rbenv/versions/3.1.2

Downloading ruby-3.1.2.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.1/ruby-3.1.2.tar.gz
Installing ruby-3.1.2...
patching file configure.ac
ruby-build: using readline from homebrew
ruby-build: using gmp from homebrew
Installed ruby-3.1.2 to /Users/mbp_arj/.rbenv/versions/3.1.2
```
[ruby-build.20221003233044.24676.log](https://github.com/rbenv/ruby-build/files/9727757/ruby-build.20221003233044.24676.log)

Catalina 10.15.7
XCode Version 12.4
Brew upgrade run

**Comment by mislav** [comment]
Thanks for sharing your workaround. I am not entirely sure what exactly causes this issue or why the patch is needed in the first place.

Are you saying that you solved the problem by reinstalling your Xcode CommandLine tools, applying the patch, or by doing both?

We can consider adding this to the troubleshooting section here: https://github.com/rbenv/ruby-build/wiki#troubleshooting

**Reply by WarthogARJ** [reply]
Hi: sorry for my delay in reply.
I'm not sure either. I've subequently switched to another MacBookPro Mid 2012, and when I setup another SSD with Brew etc, I didn't see the same issue. 

To solve my reported issue, re-installing XCode Command ools did NOT solve it.
At least just re-installing via Homebrew. But I didn't re-install XCode in entirety.

The only solution was installing the patch.

And yes, I think until someone can feed in more info to address the root cause, I think mentioning it in Troubleshooting as a work-around would be very helpful.

Thanks for your reply, and the usual great support.
